central/complianceoperator/v2/checkresults/datastore/datastore_impl.go

```diff
@@ -465,3 +468,26 @@ func (d *datastoreImpl) withCountByResultSelectQuery(q *v1.Query, countOn search
 	)
 	return cloned
 }
+
+func (d *datastoreImpl) DeleteOldResults(ctx context.Context, lastStartedTimestamp *timestamppb.Timestamp, scanRefID string, includeCurrent bool) error {
+	if scanRefID == "" || lastStartedTimestamp == nil {
+		return nil
+	}
+	lastStartedTimestampColumnName := "laststartedtime"
+	scanRefIDColumnName := "scanrefid"
+	deleteFmt := "DELETE FROM %s WHERE %s = $1 AND (%s < $2 OR %s IS NULL)"
+	if includeCurrent {
+		deleteFmt = "DELETE FROM %s WHERE %s = $1 AND (%s <= $2 OR %s IS NULL)"
+	}
+	deleteStmt := fmt.Sprintf(deleteFmt,
+		schema.ComplianceOperatorCheckResultV2TableName,
+		scanRefIDColumnName,
+		lastStartedTimestampColumnName,
+		lastStartedTimestampColumnName,
+	)
+	_, err := d.db.Exec(ctx, deleteStmt, scanRefID, protocompat.NilOrTime(lastStartedTimestamp))
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
We are not wrapping here, so maybe
```suggestion
	return err
```
?

Reply:
Done